### PR TITLE
Update maxpath in kmergraph

### PR DESCRIPTION
**Pull request by rmcolq**
This is for reference only, showing current state of work to do the following:
1. Issue #145 to average log probs over some fixed number of kmers, not all
2. Breaking KmerGraph and KmerGraphWithCoverage into separate files
3. Add tests corresponding to KmerGraphWithCoverage back in
4. Improve variable names as per comments in https://github.com/rmcolq/pandora/pull/142
5. Where obvious, improve code coverage in KmerGraphWithCoverage at the same time

So far have addressed 2 and 1, and am part way through adding tests for the new code. Still to finish this, improve code coverage and improve variable names.

### CODE REVIEW

**Comment by rmcolq** [comment]
Right, so the tests of my new functions pass, and line coverage for KmerGraphWithCoverage is up to 92.3 %. Are we happy for a merge? 
@leoisl I haven't yet added an equals operator to KmerGraphWithCoverage. Hoping you may be able to add one if it's needed...?

File: src/kmergraphwithcoverage.cpp [hunk 1]
```diff
@@ -63,67 +63,62 @@ void KmerGraphWithCoverage::set_covg(uint32_t node_id, uint32_t value, bool stra
...
 
-float KmerGraphWithCoverage::nb_prob(uint32_t j, const uint32_t &sample_id) {
-    auto k = get_covg(j, 0, sample_id) + get_covg(j, 1, sample_id);
-    //cout << "j: " << j << " " << nodes[j]->covg[0] << " + " << nodes[j]->covg[1] << " = "
-    //     << nodes[j]->covg[0] + nodes[j]->covg[1] << " = " << k << endl;
-    //cout << "nb_prob(" << nb_r << "," << nb_p << "," << k << ") = ";
-    float ret = log(pdf(boost::math::negative_binomial(nb_r, nb_p), k));
+float KmerGraphWithCoverage::nbin_prob(uint32_t node_id, const uint32_t &sample_id) {
+    auto k = get_covg(node_id, 0, sample_id) + get_covg(node_id, 1, sample_id);
+    float ret = log(pdf(boost::math::negative_binomial(negative_binomial_parameter_r, negative_binomial_parameter_p), k));
```
Comment (mbhall88) [inline src/kmergraphwithcoverage.cpp]: Hate to be a pest, but what is `ret`?

**Comment by mbhall88** [comment]
Loving these renames :heart_eyes: . Big increase in code readability - coming from someone who doesn't really understand this part of the code.

**Comment by rmcolq** [comment]
Return value. I'll update in a bit.



Sent from my Samsung Galaxy smartphone.


-------- Original message --------

**Comment by leoisl** [comment]
Hello,

sorry for the delay! I went through the whole code changes and everything is fine for me! Nice work!

> line coverage for KmerGraphWithCoverage is up to 92.3 %

That is nice, thanks a lot!

> @leoisl I haven't yet added an equals operator to KmerGraphWithCoverage. Hoping you may be able to add one if it's needed...?

Sure !

> Loving these renames heart_eyes . Big increase in code readability - coming from someone who doesn't really understand this part of the code.

Me too, it is really more understandable for me now!

**Comment by leoisl** [comment]
@rmcolq Looking at the `KmerGraphWithCoverage` definition, we declare the copy constructor with the default implementation:

https://github.com/rmcolq/pandora/blob/f7257e4096a7831409869d9681b8979349b8c7f6/include/kmergraphwithcoverage.h#L44

As there is nothing customized we should do to copy one KmerGraphWithCoverage object to the other (just copy each attribute), I think the default copy constructor is fine. Do you agree with this?

**Comment by rmcolq** [comment]
I agree about default constructors - maybe if we have a careful definition of equals operator, we can check with a test?

**Comment by leoisl** [comment]
Oh, please pardon me, ignore my previous message... I don't know why wherever it was written "equal operator" I was thinking on `operator=` and copy ctor, but not on `operator==`... I don't know why, really a misunderstanding from my part, will implement `operator==` tomorrow!

**Pull request merged**